packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.controller.js

```diff
@@ -3,26 +3,34 @@ import find from 'lodash/find';
 import first from 'lodash/first';
 import get from 'lodash/get';
 import forEach from 'lodash/forEach';
+import isEmpty from 'lodash/isEmpty';
 
 export default class FlavorsListController {
   /* @ngInject */
-  constructor(PciProjectFlavors) {
+  constructor($q, PciProjectFlavors) {
+    this.$q = $q;
     this.PciProjectFlavors = PciProjectFlavors;
   }
 
   $onInit() {
     this.isLoading = true;
-
+    this.flavorCount = this.flavorCount || 1;
     return this.getFlavors().finally(() => {
       this.isLoading = false;
     });
   }
 
   getFlavors() {
-    return this.PciProjectFlavors.getFlavors(
-      this.serviceName,
-      get(this.region, 'name'),
-    ).then((flavors) => {
+    let flavorsPromise = null;
+    if(!isEmpty(this.flavors)) {
+      flavorsPromise = this.$q.when(this.flavors);
+    } else {
+      flavorsPromise = this.PciProjectFlavors.getFlavors(
+        this.serviceName,
+        get(this.region, 'name'),
+      );
+    }
+    flavorsPromise.then((flavors) => {
```

Comment:
```suggestion
return (!isEmpty(this.flavors) ? this.$q.when(this.flavors) : this.PciProjectFlavors.getFlavors(
        this.serviceName,
        get(this.region, 'name'),
      ))
      .then(...)
```
It just depends how you prefer to write it though so it's nitpicking :)

Reply:
.then(...) need to be executed after if and else. I can not return "this.$q.when(this.flavors)" as you suggested above.